compiler/ccgtypes.nim

```diff
@@ -774,7 +774,7 @@ proc getRecordDesc(m: BModule; typ: PType, name: Rope,
     result = structOrUnion & " " & name
     result.add(getRecordDescAux(m, typ, name, baseType, check, hasField))
     let desc = getRecordFields(m, typ, check)
-    if not hasField:
+    if not hasField and m.g.graph.memberProcsPerType.len == 0:
```

Comment:
What? Why does that matter?

Reply:
wdym? To switch the test against a proc flag? or are you asking why is it necessary to test there?

Reply:
Just explain the reasoning behind this logic here please.

Reply:
Before this regression, the code looked like:
```nim
 if desc == "" and not hasField:
      result.addf("char dummy;$n", [])

```
By removing `desc` as a test by itself (hence my question on internals) broke it because `constructor` and `virtual` does write in `desc`

Reply:
I still don't get it, there is nothing type specific about the test `m.g.graph.memberProcsPerType.len == 0` which means as soon as memberProcsPerType contains something the code generation for every type would be affected.

Reply:
Apologies, dont know what I was thinking :/